pkg/controller/admissionchecks/provisioning/controller_test.go

```diff
@@ -724,3 +724,133 @@ func TestReconcile(t *testing.T) {
+				},
+				Count: ptr.To[int32](4),
+			},
+		).
+			Obj()).
+		AdmissionChecks(kueue.AdmissionCheckState{
+			Name:  "check",
+			State: kueue.CheckStatePending,
+		}, kueue.AdmissionCheckState{
+			Name:  "not-provisioning",
+			State: kueue.CheckStatePending,
+		}).
+		Obj()
+	baseConfig := &kueue.ProvisioningRequestConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "config1",
+		},
+		Spec: kueue.ProvisioningRequestConfigSpec{
+			ProvisioningClassName: "class1",
+			Parameters: map[string]kueue.Parameter{
+				"p1": "v1",
+			},
+		},
+	}
+
+	baseRequest := autoscaling.ProvisioningRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: TestNamespace,
+			Name:      "wl-check-1",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Name: "wl",
+				},
+			},
+		},
+		Spec: autoscaling.ProvisioningRequestSpec{
+			PodSets: []autoscaling.PodSet{
+				{
+					PodTemplateRef: autoscaling.Reference{
+						Name: "ppt-wl-check-1-ps1",
+					},
+					Count: 4,
+				},
+			},
+			ProvisioningClassName: "class1",
+			Parameters: map[string]autoscaling.Parameter{
+				"p1": "v1",
+			},
+		},
+	}
+	pr1Failed := baseRequest.DeepCopy()
+	pr1Failed = requestWithCondition(pr1Failed, autoscaling.Failed, metav1.ConditionTrue)
+	pr2Created := baseRequest.DeepCopy()
+	pr2Created.Name = "wl-check-2"
+
+	baseCheck := utiltesting.MakeAdmissionCheck("check").
+		ControllerName(ControllerName).
+		Parameters(kueue.GroupVersion.Group, ConfigKind, "config1").
+		Obj()
+
+	cases := map[string]struct {
+		requests   []autoscaling.ProvisioningRequest
+		wantResult map[string]*autoscaling.ProvisioningRequest
+	}{
+		"no provisioning requests": {},
+		"two provisioning requests; 1 then 2": {
+			requests: []autoscaling.ProvisioningRequest{
+				*pr1Failed.DeepCopy(),
+				*pr2Created.DeepCopy(),
+			},
+			wantResult: map[string]*autoscaling.ProvisioningRequest{
+				"check": pr2Created.DeepCopy(),
+			},
+		},
+		"two provisioning requests; 2 then 1": {
+			requests: []autoscaling.ProvisioningRequest{
+				*pr2Created.DeepCopy(),
+				*pr1Failed.DeepCopy(),
+			},
+			wantResult: map[string]*autoscaling.ProvisioningRequest{
+				"check": pr2Created.DeepCopy(),
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			workload := baseWorkload.DeepCopy()
+			relevantChecks := []string{"check"}
+			checks := []kueue.AdmissionCheck{*baseCheck.DeepCopy()}
+			configs := []kueue.ProvisioningRequestConfig{*baseConfig.DeepCopy()}
+
+			builder, ctx := getClientBuilder()
+
+			builder = builder.WithObjects(workload)
+			builder = builder.WithStatusSubresource(workload)
+
+			builder = builder.WithLists(
+				&autoscaling.ProvisioningRequestList{Items: tc.requests},
+				&kueue.ProvisioningRequestConfigList{Items: configs},
+				&kueue.AdmissionCheckList{Items: checks},
+			)
+
+			k8sclient := builder.Build()
+			recorder := &utiltesting.EventRecorder{}
+			controller, _ := NewController(k8sclient, recorder)
```

Comment:
Do not obscure the error.

Call `t.Fatalf` if there is an error that doesn't allow us to continue with the rest of the test.

Reply:
Done, also added it to `TestReconcile` which existed before (and from which I copied this bit).